bucket.go

```diff
@@ -43,7 +43,7 @@ type Bucket interface {
 	WriteCasWithXattr(k string, xattrKey string, exp int, cas uint64, v interface{}, xv interface{}) (casOut uint64, err error)
 	GetWithXattr(k string, xattrKey string, rv interface{}, xv interface{}) (cas uint64, err error)
 	DeleteWithXattr(k string, xattrKey string) error
-	WriteUpdateWithXattr(k string, xattrKey string, exp int, callback WriteUpdateWithXattrFunc) (casOut uint64, err error)
+	WriteUpdateWithXattr(k string, xattrKey string, exp int, previousValue []byte, previousXattr []byte, cas uint64, callback WriteUpdateWithXattrFunc) (casOut uint64, err error)
```

Comment:
Could this be wrapped up in a struct called Previous with these two fields?  That would reduce some of the cognitive load when looking at this (and calls)

The struct could be in the sgbucket package and therefore be available to all packages that depend on this.